### ISSUE TITLE
Xori install error on Windows 10

### ISSUE THREAD

**Issue by SeattleForensics** [endgameinc/xori#6]
I've attempted to install Xori on two versions of Windows 10; both the latest and greatest v 1803 as well as the previous version (1709 I believe). Per the requirements:
-Installed Rust from https://static.rust-lang.org/dist/rust-1.28.0-x86_64-pc-windows-gnu.msi
-Downloaded the zip from https://github.com/endgameinc/xori
-Tried installing in CMD window, both with and without Admin privileges using "cargo build --release".
-Also tried installing with the same command in a PowerShell window.

In all cases, I got the attached error message.

Can you help me troubleshoot this?
![install-error](https://user-images.githubusercontent.com/24799443/44016216-0882d646-9e89-11e8-846a-19f32afa1da3.png)

**Comment by securedorg** [comment]
Have you also installed GCC in windows? The error message said it can’t find “make” in the environment. If you already have visual studio installed I recommend the rust msvc build and not gnu.

**Comment by d1maz** [comment]
Hey @SeattleForensics , I've just build it successfully on the win10 VM from https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/

try this:
- download **x86_64-pc-windows-gnu** (_"rustup-init.exe"_) from here https://www.rust-lang.org/en-US/other-installers.html
- run it, if your environment is missing the **Build Tools for Visual Studio 2017** you will get the link from the cli output
- download & install the **Build Tools for Visual Studio 2017** from here https://visualstudio.microsoft.com/downloads/ (at the bottom, expand "Tools for Visual Studio 2017")
- run _"rustup-init.exe"_ again & after this step you can download **xori** & build it
- tinker around & enjoy!

![xori_build](https://user-images.githubusercontent.com/9216754/44043108-dc61549a-9ed6-11e8-928b-a4e0ec47db39.png)

**Comment by rseymour** [comment]
Thanks @d1maz i will try to get this comment into a wiki here or something so folks can get moving if dependency heck is giving them headaches.

**Comment by Frenchy7** [comment]
Hi, after installing xori, how to start malware analysis ?

**Comment by SeattleForensics** [comment]
Thanks everyone. I tried the msvc build this time. The combination that ended up working for that was to not only install Build Tools for Visual Studio 2017, but also Visual Studio itself with the "Desktop Development with C++" workload/package. Installing the build tools alone still left out key components such as "link.exe", which was necessary for the compilation to complete.  Installing Visual Studio without any additional components I believe also failed, but once I chose the "Desktop development with C++" workload, it worked.  

One installation issue I can see that may be confusing for some perspective users is the choice of what to install when you run the Build Tools for Visual Studio 2017. Once you select the Community Edition, there are 18 separate boxes that can be checked. Steering people to the "Desktop Development with C++" option may be helpful, as I had to choose that one as well to get successful compilation. There may be a more minimal set of things needed to compile (which is what I was looking for to avoid bloat because the download was several GB). 

I tried the x86_64-pc-windows-gnu version of rust previous to the above success, and had installed the Build Tools for Visual Studio 2017, but compilation still failed, even on an up to date Windows 10 box, build 1803. I'm curious what was different between your successful test and my failed attempt, but in the end I don't think any of us have the spare time to track such things down when there was a different solution.  My thought behind posting this was to bring to light installation issues that others may run into, which could potentially hurt the adoption rate of trying out Xori.  Ultimately, releasing precompiled binaries would make it easiest on people, but I'm sure there must have been reasons that was not done.

Thanks again.
